.github/workflows/valgrind.yml

```diff
@@ -31,20 +33,58 @@ jobs:
           sudo apt-get install libz3-dev libz3-4 libz3-4-dbgsym
           sudo apt-get install libc6-dbg-amd64-cross
 
-      - name: Build cppcheck
-        run: |
-          CXXFLAGS="-O1 -g" make -j$(nproc) USE_Z3=yes HAVE_RULES=yes MATCHCOMPILER=yes
-
-      - name: Build test
+      - name: Build cppcheck debug
         run: |
-          CXXFLAGS="-O1 -g" make -j$(nproc) testrunner USE_Z3=yes HAVE_RULES=yes MATCHCOMPILER=yes
-
-      - name: Run valgrind
+          CXXFLAGS="-O1 -g" make -j$(nproc) USE_Z3=yes HAVE_RULES=yes MATCHCOMPILER=yes CPPFLAGS="-DCHECK_INTERNAL" VERIFY=1
+     
+      - name: Run valgrind on cppcheck self-check
         run: |
-          valgrind --error-limit=yes --leak-check=full --num-callers=50 --show-reachable=yes --track-origins=yes --suppressions=valgrind/testrunner.supp --gen-suppressions=all --log-fd=9 --error-exitcode=42 ./testrunner TestGarbage TestOther TestSimplifyTemplate 9>memcheck.log
-          cat memcheck.log
+          valgrind --error-limit=yes --leak-check=full --num-callers=50 --show-reachable=yes --track-origins=yes --gen-suppressions=all --log-fd=9 --error-exitcode=42 ./cppcheck --enable=style,performance,portability,warning,internal --exception-handling --debug-warnings cli lib 9>memcheck_cppcheck.log
```

Comment:
You are lacking `-D__CPPCHECK__ --error-exitcode=1 --inline-suppr --suppressions-list=.travis_suppressions --library=cppcheck-lib -Ilib -Iexternals/simplecpp/ -Iexternals/tinyxml2/ -Icli --inconclusive` for the Cppcheck execution.

Reply:
Forgot another one. Still lacking `--template=selfcheck` - please rebase to get that and having the build actually fail on an error as https://github.com/danmar/cppcheck/pull/3023 was merged.

Also after changes related to https://github.com/danmar/cppcheck/pull/2996 and https://github.com/danmar/cppcheck/pull/3046 have been merged Cppcheck should be running much faster.

We should put the options for the self-check in an external file. Those parameters are (largely) used in multiple places right now and we will most likely have more usages added in the future. With a single place we would make sure all of them are actually in sync. Just having a variable somewhere in the actions would be enough as a start.

When I want to test it locally I always have to copy it out of the YAML file. So having it available from the command-line would be helpful as well.